### PR TITLE
Add getTenant operation to TenantManager.

### PR DESCRIPTION
**Pull request by micahstairs**
This pull request add getTenant to the TenantManager class. I've added the relevant unit tests to FirebaseUserManagerTest (based off of the getUser unit tests). This is part of the initiative to adding multi-tenancy support (see issue #332).

REST API: https://identitytoolkit.googleapis.com/$discovery/rest?version=v2

### CODE REVIEW

File: src/main/java/com/google/firebase/auth/FirebaseUserManager.java [hunk 1]
```diff
@@ -226,6 +228,16 @@ UserImportResult importUsers(UserImportRequest request) throws FirebaseAuthExcep
     return new UserImportResult(request.getUsersCount(), response);
   }
 
+  Tenant getTenantById(String tenantId) throws FirebaseAuthException {
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/FirebaseUserManager.java]: I think `getTenant` is descriptive enough. We have several different ways of fetching user accounts (by UID, by email etc), but we don't have that requirement here.

File: src/main/java/com/google/firebase/auth/FirebaseUserManager.java [hunk 2]
```diff
@@ -226,6 +228,16 @@ UserImportResult importUsers(UserImportRequest request) throws FirebaseAuthExcep
...
+    GenericUrl url = new GenericUrl(tenantMgtBaseUrl + "/tenants:get");
+    url.put("name", tenantId);
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/FirebaseUserManager.java]: This doesn't quite look right. The URL path should be something like `v2/projects/{projectsId}/tenants/{tenantsId}`. That's what we use in other SDKs, and also what's listed in the auto-generated discovery doc.

File: src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java [hunk 3]
```diff
@@ -430,6 +430,28 @@ public void testImportUsersLargeList() {
...
+    TestResponseInterceptor interceptor = initializeAppForUserManagement(
+        TestUtils.loadResource("getTenant.json"));
+    Tenant tenant = FirebaseAuth.getInstance().getTenantManager().getTenantAsync("TENANT_1").get();
+    checkTenant(tenant, "TENANT_1");
+    checkRequestHeaders(interceptor);
+  }
+
+  @Test
+  public void testGetTenantWithNotFoundError() throws Exception {
+    initializeAppForUserManagement("{}");
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java]: I believe this endpoint sends a 404 (or similar error response) when the tenant is not found. Lets update the test case accordingly. See https://github.com/firebase/firebase-admin-go/blob/e088f55036154f0cccf661992f854abd8fb1333f/auth/tenant_mgt_test.go#L1096-L1100

Comment (micahstairs) [reply src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java]: I don't think I follow. The Node implementation suggests I check for a response missing a 'name' (which is the tenantId): https://github.com/firebase/firebase-admin-node/blob/2b952d43fd8784151ebdb60b65e88b9fd1888ccf/src/auth/auth-api-request.ts#L1463.

File: src/main/java/com/google/firebase/auth/FirebaseUserManager.java (hunk 2 above)
Comment (micahstairs) [reply src/main/java/com/google/firebase/auth/FirebaseUserManager.java]: Thanks! Fixed.

File: src/main/java/com/google/firebase/auth/FirebaseUserManager.java (hunk 1 above)
Comment (micahstairs) [reply src/main/java/com/google/firebase/auth/FirebaseUserManager.java]: Done.

**Comment by micahstairs** [comment]
I think I'm also going to assert that the URL of the outgoing request is correct, like you suggested on my other PR.

**Comment by micahstairs** [comment]
> I think I'm also going to assert that the URL of the outgoing request is correct, like you suggested on my other PR.

Done.

File: src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java (hunk 3 above)
Comment (hiranya911) [reply src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java]: That's just some additional validation they do on the "successful" responses. On missing tenants, the backend sends a 404 NOT_FOUND error.

File: src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java [hunk 4]
```diff
@@ -437,19 +437,23 @@ public void testGetTenant() throws Exception {
     Tenant tenant = FirebaseAuth.getInstance().getTenantManager().getTenantAsync("TENANT_1").get();
     checkTenant(tenant, "TENANT_1");
     checkRequestHeaders(interceptor);
+    checkUrl(interceptor,
+        "https://identitytoolkit.googleapis.com/v2/projects/test-project-id/tenants/TENANT_1");
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java]: Nit: Define the URL prefix (up to the /tenants portion) in a constant.

File: src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java [hunk 5]
```diff
@@ -430,6 +430,32 @@ public void testImportUsersLargeList() {
...
+        TestUtils.loadResource("getTenant.json"));
+    Tenant tenant = FirebaseAuth.getInstance().getTenantManager().getTenantAsync("TENANT_1").get();
+    checkTenant(tenant, "TENANT_1");
+    checkRequestHeaders(interceptor);
+    checkUrl(interceptor,
+        "https://identitytoolkit.googleapis.com/v2/projects/test-project-id/tenants/TENANT_1");
+  }
+
+  @Test
+  public void testGetTenantWithNotFoundError() throws Exception {
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java]: Compare with https://github.com/firebase/firebase-admin-go/blob/e088f55036154f0cccf661992f854abd8fb1333f/auth/tenant_mgt_test.go#L1151-L1161

File: src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java (hunk 3 above)
Comment (micahstairs) [reply src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java]: Done.

File: src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java (hunk 4 above)
Comment (micahstairs) [reply src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java]: Done.

**Comment by micahstairs** [comment]
I had to rebase since the other PR was submitted first. Hopefully I did that right.

**Comment by micahstairs** [comment]
It took a lot of work to get this rebased properly, but I learned a lot about git in the process :)

The individual commits are bit messed up (the code wasn't always in a state where it could be compiled after each commit), however, it's all about to get squashed anyway.

**Pull request merged**